ibverbs/src/lib.rs

```diff
@@ -183,6 +183,54 @@ impl<'d> From<&'d *mut ffi::ibv_device> for Device<'d> {
     }
 }
 
+/// A Global unique identifier for ibv.
+///
+/// This struct acts as a rust wrapper for GUID value represented as `__be64` in
+/// libibverbs. We introduce this struct, because u64 is stored in host
+/// endianness, whereas ibverbs stores GUID in network order (big endian).
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Default, Copy, Clone, Debug, Eq, PartialEq, Hash)]
+#[repr(transparent)]
+pub struct Guid {
+    raw: [u8; 8],
+}
+
+impl Guid {
+    /// Upper 24 bits of the GUID are OUI (Organizationally Unique Identifier,
+    /// http://standards-oui.ieee.org/oui/oui.txt). The function returns OUI as
+    /// a 24-bit number inside a u64 number.
```

Comment:
```suggestion
    /// a 24-bit number inside a u32.
```
I think this is what you mean?